.claude/skills/docx/scripts/document.py

```
@@ -0,0 +1,1276 @@
+            raise ValueError(f"Element must be w:r or w:p, got {elem.nodeName}")
+
+
+def _generate_hex_id() -> str:
+    """Generate random 8-character hex ID for para/durable IDs.
+
+    Values are constrained to be less than 0x7FFFFFFF per OOXML spec:
+    - paraId must be < 0x80000000
+    - durableId must be < 0x7FFFFFFF
+    We use the stricter constraint (0x7FFFFFFF) for both.
+    """
+    return f"{random.randint(1, 0x7FFFFFFE):08X}"
```

Ruff (0.14.14)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)